### PR TITLE
12 file and folder retrieval from database

### LINKED ISSUE #12: File and Folder retrieval from database
As a user I want to retrieve files from the database and open folders. Also I want to see the last chaged files.
Therefore as a developer I need a method that let's me access these files easily.

Acceptance criteria:
2 Functions

```
getLastChangedFiles()
//Searches for the last changed files
 return FileCardType[ ]
```
```
getFolderContents()
//Gets the content of the 
return (FolderCardType[ ], FileCardType[ ])
```

### PR DESCRIPTION
**Pull request by cma5**
Closes #12

Unfortunately the expo app can't use the systems localhost, since it has it has its own localhost.
Therefore we need to access Nextcloud via the computers network IP.
If you found the IP, adjust the IP Variable in /utils/ServerRequests.tsx

```
const machineIP = "[YOU.ROW.NIP.69]";
```

Additionally, Nextcloud out of the box, only allows for access via "localhost", therefore we need to allow our computers IP in nextcloud. 
```bash
runuser --user www-data -- /usr/local/bin/php /var/www/html/occ config:system:set trusted_domains 2 --value=[YOU.ROW.NIP.69]
```

Also authentication is not implemented properly yet, therefore you need to use the nextcloud admin credentials specified in [.env](https://github.com/lern-fair-file-sharing/backend/blob/master/.env)

### CODE REVIEW

File: app/components/FolderCard.tsx [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 
 
 const FolderCard = (props: FolderNavigationType) => {
-    const handleFolderPress = () => {
+    const handleFolderPress = async() => {
         props.navigation.push("FolderContentScreen", { folderURL: props.folderURL, folderName: props.folderName });
+
+        //const content = await getFolderContent("/remote.php/dav/files/testuser");
+        //const results = await searchLatestFiles();
+        //console.log(content, results);
```
Comment (theopfr) [inline app/components/FolderCard.tsx]: Please remove those unused test calls.

File: app/components/FolderContentScreen.tsx [hunk 2]
```diff
@@ -14,40 +18,34 @@ export type RootStackParamList = {
...
-                filePreviewURL: "https://i.pinimg.com/736x/d8/27/a3/d827a3a7acd4f727f8f4f2c45e2cb309.jpg",
-                tags: ["mathe", "joe"],
-                lastModified: "l3h"
+    const [allFolders, setAllFolders] = useState<FolderCardType[]>( [] as FolderCardType[]);
+    const [allFiles, setAllFiles] = useState<FileCardType[]>( [] as FileCardType[]);
+
+    useEffect(() => {
+        const fetchFolderContent = async () => {
+            const content = await getFolderContent(props.route.params?.folderURL);
+            //console.log(content);
```
Comment (theopfr) [inline app/components/FolderContentScreen.tsx]: Also remove this comment.

**Review by theopfr — APPROVED** [review]
Markus and I did some pair programming so from my side is everything clear already. Just remove the unused comment please before you merge.

File: app/components/FolderCard.tsx (hunk 1 above)
Comment (cma5) [reply app/components/FolderCard.tsx]: removed

File: app/components/FolderContentScreen.tsx (hunk 2 above)
Comment (cma5) [reply app/components/FolderContentScreen.tsx]: remove

**Pull request merged**